datastructures/search_trees/binary_search_tree.py

```diff
@@ -0,0 +1,109 @@
+from datastructures.search_trees.search_tree import SearchTree, Node, T, V
+
+
+class BinarySearchTree(SearchTree):
+    def search(self, search_key: T, current_node: Node = None) -> tuple[Node]:
+        if current_node is None:
+            current_node = self.root
```

Comment:
```suggestion
        current_node = current_node or self.root
```

Reply:
That looks good, didn't know this is a thing